src/module.c

```diff
@@ -181,6 +185,11 @@ int GetSingleDocumentCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int
     return RedisModule_ReplyWithError(ctx, "Unknown Index name");
   }
 
+  if (IsEnterprise() && !ACLUserMayAccessIndex(ctx, sctx->spec)) {
```

Comment:
Maybe better to unified this to a function, `checkEnterpriseACL`?

Reply:
Yeah sounds good